### ISSUE TITLE
Why was the ORM removed?

### ISSUE THREAD

**Issue by fernandocamargoai** [thingsdb/python-thingsdb#37]
I discovered ThingsDB because of this blog post: https://medium.com/swlh/keep-data-in-sync-between-multiple-services-using-thingsdb-d94a4aa0e9fc

I have that usecase of syncing a collection of objects between different services and was planning to use ThingsDB for that. Then, I noticed that there's no `thingsdb.model` anymore and after further looking, I've seen that the ORM was removed in version 1.0.0.rc1. So, I wonder why it was removed and what would be the alternative API to be used in that case (if any)?

**Comment by joente** [comment]
Hi @fernandocamargoai, first of all thanks for looking at ThingsDB!

Since the time of writing the blog post, many changes have been made to ThingsDB.
A short history: we started writing both ThingsDB and the ORM Python client simultaneous but after some time we noticed it became more and more difficult to keep the ORM layer in sync with the changes in ThingDB.
The ORM layer (as we had it back in the days) did a lot of *magic* and as a result it was not always clear to the programmer what happened with the Python Objects which were being synchronized.
It also became clear that writing the same ORM layer for other languages (than Python) would be very difficult and hard to maintain. 

But, we found a very neat solution! Instead of *watching* things directly, we introduced the [room type](https://github.com/thingsdb/ThingsDB/issues/209) in ThingsDB. This has many benefits:

1. With rooms it easy to understand for the programmer what happens with your Python Objects.
2. Using rooms it will be an easy task to implement a ThingsDB connector for another programming language.
3. Much easier to maintain the connectors as we can keep the room API equal no matter what we change in ThingsDB.
4. More stable. The ORM layer required a lot of code which was difficult to understand while the current connector is pretty clean.

Unfortunately, we did not yet update the blog post (yet), but this morning I did wrote another [bookstore example](https://github.com/thingsdb/python-thingsdb/blob/main/examples/bookstore/bookstore.py) which explains how to use the room type. You might also want to look at the [room documentation](https://docs.thingsdb.net/v1/data-types/room/emit/).
